### ISSUE TITLE
Following the guide here to upload a static site can result in leaking credentials

### ISSUE THREAD

**Issue by pdylanross** [google-github-actions/upload-cloud-storage#215]
### TL;DR
When using this action to upload site content to a bucket, if the site content lives at the root of the repository, then this action will also upload the gcloud credentials file. 

**Expected behavior**
Not uploading credential files.

**Observed behavior**
Job output:
```
 Uploading file: page.html to gs://bucket/page.html
Uploading file: other_page.html to gs://bucket/other_page.html
Uploading file: gha-creds-xyzxyzxyxz.json to gs://bucket/gha-creds-xyzxyzxyxz.json
```

### Reproduction

**Action YAML**
```yaml
name: Deploy

on:
  push:
    branches:
      - master

jobs:
  deploy:
    name: Deploy
    runs-on: ubuntu-latest
    permissions:
      contents: read
      id-token: write
    steps:
      - uses: actions/checkout@v2
      - id: auth
        uses: google-github-actions/auth@v0
        with:
          workload_identity_provider: ${{ secrets.GCLOUD_WORKLOAD_IDENTITY_PROVIDER }}
          service_account: ${{ secrets.GCLOUD_SERVICE_ACCOUNT }}
      - id: upload-artifacts
        uses: google-github-actions/upload-cloud-storage@v0
        with:
          path: ./
          destination: bucket
```
**Additional information**
The repo isn't public, but it's more or less a few raw html, css, js, and image files checked in. The root of the repo is the root of the site. This issue isn't blocking me and I'm going to find ways to work around it, but the bug serves just as much of a warning to others how easy this action makes it to accidentally leak credentials.

**Comment by pdylanross** [comment]
For anyone else bumping into this, a quick solution is to checkout the code into a child path in the workspace and then upload from there:

```yaml
jobs:
  deploy:
    name: Deploy
    steps:
      - uses: actions/checkout@v2
        with:
          path: ./site
      ...
      - id: upload-artifacts
        uses: google-github-actions/upload-cloud-storage@v0
        with:
          path: ./site
          destination: bucket
```

**Comment by sethvargo** [comment]
Hi @pdylanross 

Please see the instructions in the [`auth` troubleshooting guide for bundled credentials](https://github.com/google-github-actions/auth/blob/main/docs/TROUBLESHOOTING.md#dirty-git-or-bundled-credentials).

@bharathkkb I thought we used to process the gcloudignore/gitignore, but I can't seem to find where we do that in the code.

**Comment by sethvargo** [comment]
Agreed - let's start with skipping `gha-creds-*`.
